File: LeetCode/07 - Reverse integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:

        # Setup for handling negative numbers
        is_negative: bool = str(x).startswith("-")

        # Handle integers larger than 2^31 (2,147,483,647) or less than -2^31 (-2,147,483,647)
        if -2147483647 > x or x > 2147483647:
            return 0

        # Split the number into an array and strip any negative symbols
        digits: list[str] = [n for n in str(x).strip("-")]

        reversed_number = int("".join(digits[::-1]))
        reversed_number *= -1 if is_negative else 1

        # If the reversed value is greater than 2^31 or less than -2^31
        if -2147483647 > reversed_number or reversed_number > 2147483647:
            return 0
        else:
            return reversed_number

File: LeetCode/07 - Reverse integer/solution.swift
class Solution {
    func reverse(_ x: Int) -> Int {

        // Handle negative numbers
        var is_negative: Bool = String(x).hasPrefix("-")

        // Handle integers greater than or less than 2^31 and -2^31
        if -2147483647 > x || x > 2147483647 {
            return 0
        }

        // Array for all the individual digits of the number
        var digits: Array<String> = []

        for number in String(x) {
            // If the variable is a negative symbol, skip past it
            if number == "-" {
                continue
            } else {
                digits.append(String(number))
            }
        }

        // Reverse the digits array and then join the array into an integer
        digits.reverse()
        var reversed_number: Int = Int(digits.joined(separator: ""))!

        // Make the reversed_number negative if the input number was negative
        if is_negative {
            reversed_number *= -1
        }

        // Handle integers greater than or less than 2^31 and -2^31
        if -2147483647 > reversed_number || reversed_number > 2147483647 {
            return 0
        }

        return reversed_number

    }
}